sdk/core/Azure.Core/src/DynamicData/AllowListConverterFactory.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core.Dynamic;
+
+namespace Azure.Core.Json
+{
+    internal class AllowListConverterFactory : JsonConverterFactory
+    {
+        public static readonly AllowListConverterFactory Default = new();
+
+        public override bool CanConvert(Type typeToConvert)
+        {
+            return !IsAllowedType(typeToConvert);
+        }
+
+        public static bool IsAllowedType(Type type)
+        {
+            if (IsAllowedKnownType(type))
+            {
+                return true;
+            }
+
+            return IsAllowedPoco(type, new List<Type>());
+        }
+
+        private static bool IsAllowedPoco(Type type, List<Type> ancestorTypes)
+        {
+            if (!HasPublicParameterlessConstructor(type) && !IsAnonymousType(type))
```

Comment:
We'll want to agree on this definition of how to identify a "simple POCO" we're allowing to serialize.

Reply:
primitive :- int, byte, short, float, etc. 
value :- primitive | object | array
object :- (name,value)*
array :- value*

Reply:
On that note, there's `Type.IsPrimitive` we should use (or something like that).